torch_xla/csrc/ops/dynamic_ir.h

```diff
@@ -53,6 +53,21 @@ class SizeNode : public XlaNode, public torch::lazy::DimensionNode {
   mutable int64_t runtime_size_;
 };
 
+class SizeEq : public XlaNode, public torch::lazy::DimensionNode {
+ public:
+  SizeEq(torch::lazy::Value a, torch::lazy::Value b);
+  int64_t getDynamicValue() const override;
+  int64_t getStaticValue() {
+    TORCH_CHECK(false, "Comparison operators should be using getDynamicValue");
```

Comment:
I guess the use case for the SizeEq node is that users does a `tensor1.shape[0] == tensor2.shape[0]` or `tensor1.shape[0] == tensor1.shape[1]` or something similar, right?

Reply:
any two tensor shapes can be compared.